dronegen/push.go

```diff
@@ -0,0 +1,134 @@
+		`git submodule update --init --recursive webassets || true`,
+		`rm -f /root/.ssh/id_rsa`,
+	}
+	if fips {
+		commands = append(commands, `if [[ "${DRONE_TAG}" != "" ]]; then echo "${DRONE_TAG##v}" > /go/.version.txt; else egrep ^VERSION Makefile | cut -d= -f2 > /go/.version.txt; fi; cat /go/.version.txt`)
+	}
+	return commands
+}
+
+// pushBuildCommands generates a list of commands for Drone to build an artifact as part of a push build
+func pushBuildCommands(b buildType) []string {
+	commands := []string{
+		`apk add --no-cache make`,
+		`chown -R $UID:$GID /go`,
+		`cd /go/src/github.com/gravitational/teleport`,
+	}
+	if b.fips {
+		commands = append(commands,
+			`export VERSION=$(cat /go/.version.txt)`,
+		)
+	}
+	commands = append(commands,
+		fmt.Sprintf(`make -C build.assets %s`, releaseMakefileTarget(b)),
+	)
+	return commands
+}
+
+// pushPipelines builds all applicable push pipeline combinations
+func pushPipelines() []pipeline {
+	var ps []pipeline
+	for _, arch := range []string{"amd64", "386", "arm", "arm64"} {
+		for _, fips := range []bool{false, true} {
+			if (arch == "386" || arch == "arm") && fips {
+				// FIPS mode not supported on i386/ARM
+				continue
+			}
+			ps = append(ps, pushPipeline(buildType{os: "linux", arch: arch, fips: fips}))
+		}
+	}
+	// Only amd64 Windows is supported for now.
+	ps = append(ps, pushPipeline(buildType{os: "windows", arch: "amd64"}))
+	return ps
+}
+
+// pushPipeline generates a push pipeline for a given combination of os/arch/FIPS
+func pushPipeline(b buildType) pipeline {
+	if b.os == "" {
+		panic("b.os must be set")
+	}
+	if b.arch == "" {
+		panic("b.arch must be set")
+	}
+
+	pipelineName := fmt.Sprintf("push-build-%s-%s", b.os, b.arch)
+	pushEnvironment := map[string]value{
+		"UID":    value{raw: "1000"},
+		"GID":    value{raw: "1000"},
+		"GOPATH": value{raw: "/go"},
+		"OS":     value{raw: b.os},
+		"ARCH":   value{raw: b.arch},
+	}
+	if b.fips {
+		pipelineName += "-fips"
+		pushEnvironment["FIPS"] = value{raw: "yes"}
+	}
+
+	p := newKubePipeline(pipelineName)
+	p.Environment = map[string]value{
+		"RUNTIME": goRuntime,
+		"UID":     value{raw: "1000"},
+		"GID":     value{raw: "1000"},
+	}
+	p.Trigger = triggerPush
+	p.Workspace = workspace{Path: "/go"}
+	p.Volumes = dockerVolumes()
+	p.Services = []service{
+		dockerService(),
+	}
+	p.Steps = []step{
+		{
+			Name:  "Check out code",
+			Image: "docker:git",
+			Environment: map[string]value{
+				"GITHUB_PRIVATE_KEY": value{fromSecret: "GITHUB_PRIVATE_KEY"},
+			},
+			Commands: pushCheckoutCommands(b.fips),
+		},
+		{
+			Name:        "Build artifacts",
+			Image:       "docker",
+			Environment: pushEnvironment,
+			Volumes:     dockerVolumeRefs(),
+			Commands:    pushBuildCommands(b),
+		},
+		{
+			Name:  "Send Slack notification",
+			Image: "plugins/slack",
+			Settings: map[string]value{
+				"webhook": value{fromSecret: "SLACK_WEBHOOK_DEV_TELEPORT"},
```

Comment:
nit: `value` as type is redundant here:
```go
"webhook": {fromSecret: "SLACK_WEBHOOK_DEV_TELEPORT"},
```

Reply:
@awly This was a thing you introduced in the first prototype of `dronegen` I think - any thoughts?